processing/src/main/java/io/druid/query/DefaultQueryMetrics.java

```diff
@@ -66,15 +65,6 @@ public void queryType(QueryType query)
   }
 
   @Override
-  public void interval(QueryType query)
```

Comment:
For all existing, single-source query types, emitting dimensions shouldn't change, i. e. it should still be "dataSource" and "interval". For the new join query type, emitted dimensions could be different. I think the most useful for users is filtering metrics by data source alone, i. e. it should still emit multi-value "dataSource" dimension.

"dataSourceAndInterval" seems too "specific" to me. Somebody may need it, however I cannot imagine in which case, but most users either won't need it, or need it in some different form. So I suggest to add hook `dataSourcesAndIntervals()` to `QueryMetrics` interface, but not emitting anything except multi-value "dataSource" dimension in `DefaultQueryMetrics`.